server/graphql/utils/request.ts

```diff
@@ -20,19 +22,61 @@ const getRequestsById = (requestIds, dataSources) => {
   return requestIds.map(id => dataSources.requests.getById(id))
 }
 
-const updateRequestHelper = (request, dataSources): Promise<ServerResponseInterface> => {
+const updateRequestHelper = async (request, dataSources): Promise<Document> => {
   if(!request.id) {
     throw new UserInputError('Missing argument value', { argumentName: 'id' })
   }
-  const requestTypeId = dataSources.requests.getById(request.id.toString()).requestType
-  return dataSources.requests.update(request)
-    .then(res => {
-      updateRequestTypeHelper({"id": requestTypeId}, dataSources)
-      return res
-    })
-    .catch(error => {
-      return error
-    })
+  const session = await mongoose.startSession()
+  try {
+    session.startTransaction()
+    const currentRequest = dataSources.requests.getById(request.id.toString())
+    const requestTypeId = currentRequest.requestType.toString()
+    const res = await dataSources.requests.update(request, session)
+    if(request.fulfilled === true && currentRequest.fulfilled === false) {
+      request.dateFulfilled = Date.now()
+    }
+    if(request.requestType) {
+      const oldRequestType = dataSources.requestTypes.getById(requestTypeId)
+      const newRequestType = dataSources.requestTypes.getById(request.requestType.toString())
+      oldRequestType.requests = oldRequestType.requests.filter(id => !id.equals(request.id))
+      newRequestType.requests.push(request.id)
+      await updateRequestTypeHelper({"id": requestTypeId, "requests": oldRequestType.requests}, dataSources)
+      await updateRequestTypeHelper({"id": request.requestType.toString(), "requests": newRequestType.requests}, dataSources)
+      request.requestType = Types.ObjectId(request.requestType)
+    }
+    else {
+      await updateRequestTypeHelper({"id": requestTypeId}, dataSources)
+    }
+    let a = false
```

Comment:
LOL i think im missing something. what's `a` for?

Reply:
oops that shouldn't be there lol